apps/src/aichat/types.ts

```diff
@@ -31,7 +21,52 @@ export const Status = AiTutorInteractionSaveStatus;
 export const PII = [Status.EMAIL, Status.ADDRESS, Status.PHONE];
 
 export interface AichatLevelProperties extends LevelProperties {
+  // --- DEPRECATED - used for old AI Chat
   systemPrompt: string;
   botTitle?: string;
   botDescription?: string;
+  // ---
+
+  /** Initial AI customizations set by the level. */
+  initialAiCustomizations?: LevelAiCustomizations;
+}
+
+/** AI customizations for student chat bots */
+export interface AiCustomizations {
+  botName: string;
+  temperature: number;
+  systemPrompt: string;
+  retrievalContexts: string[];
+  modelCardInfo?: ModelCardInfo;
 }
+
+/** Chat bot Model Card information */
+export interface ModelCardInfo {
+  description: string;
+  intendedUse: string;
+  limitationsAndWarnings: string;
+  testingAndEvaluation: string;
+  askAboutTopics: string[];
+}
+
+// Visibility for AI customization fields set by levelbuilders.
+type Visibility = 'hidden' | 'readonly' | 'editable';
+
+/**
+ * Level-defined AI customizations for student chat bots set by levelbuilders on the level's properties.
+ * Each field is the same as AiCustomizations, but with an additional visibility property.
+ */
+export type LevelAiCustomizations = {
+  [key in keyof AiCustomizations]: {
+    value: AiCustomizations[key];
+    visibility: Visibility;
+  };
+} & {
+  /** The initial panel to show when displaying the level */
+  initialPanel?: 'edit' | 'presentation';
```

Comment:
Is this being assigned the mode of the `initialPanel`, i.e., edit mode vs presentation mode?

Reply:
Yeah, we want a way for levelbuilders to specify which panel (edit/presentation) should show when first loading the level.

Reply:
What do you think of `initialPanelMode`?

Reply:
Oh interesting. Thanks for explaining! Then keeping as is makes sense.